<launch>
    <node pkg="finaltask_rxy" type="odom_tf2_broadcaster" output="screen" name="odom_tf2_broadcaster">
        <param name="file_path" value="$(find finaltask_rxy)/csv/robot1.csv"/>
        <param name="odom_id" value="odom"/>
        <param name="base_link_id" value="base_link"/>
        <param name="goal_id" value="goal"/>
    </node>

    <node pkg="finaltask_rxy" type="odom_tf2_listener" name="odom_listener" output="screen">
        <param name="base_link_id" value="base_link"></param>
        <param name="odom_id" value="odom"></param>
        <param name="goal_id" value="goal"></param>
    </node>
    <node pkg="ypspur_ros" type="ypspur_ros" name="ypspur_ros" required="true">
        <param name="param_file" value="/home/ruan-x/researches/programs/platform/yp-robot-params/robot-params/beego.param"/>
        <!--<param name="port" value="/dev/serial/by-id/usb-T-frog_project_T-frog_Driver-if00"/>-->
        <param name="port" value="/dev/ttyACM1"/>
        <param name="compatible" value="1"/>
        <param name="odom_id" value="odom"/>
        <param name="base_link_id" value="base_link"/>
    </node>
    <!-- urg -->
    <node name="hokuyo" pkg="hokuyo_node" type="hokuyo_node" respawn="false" output="screen">
      <param name="serial_port" value="/dev/ttyACM0" />
      <param name="angle_min" value="-1.5707963" />
      <param name="angle_max" value="1.5707963" />
    </node>

    <node pkg="tf" type="static_transform_publisher" name="laser_transform"
    args=" 0 0 0.2 0 0 0 base_link laser 100" />
    
    <!-- joycon -->
    <!-- <node pkg="joy" name="joy_node" type="joy_node"/>
    <node pkg="yamasemi_tf2" name="ControllerNode" type="joy_controller"/> -->
</launch>
